components/camera/transformpipeline/pipeline.go

```diff
@@ -84,21 +87,48 @@ func (cfg *transformConfig) Validate(path string) ([]string, error) {
 	return deps, nil
 }
 
+type streamCamera struct {
+	camera.Camera
+	vs gostream.VideoSource
+}
+
+func (sc *streamCamera) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
+	if sc.vs != nil {
+		return sc.vs.Stream(ctx, errHandlers...)
+	}
+	return sc.Stream(ctx, errHandlers...)
+}
+
+// streamCameraFromCamera is a hack to allow us to use Stream to pipe frames through the pipeline
+// and still implement a camera resource.
+// We prefer this methodology over passing Image bytes because each transform desires a image.Image over
+// a raw byte slice. To use Image would be to wastefully encode and decode the frame multiple times.
```

Comment:
It would also make this PR so hard because every transform would be need to be re-written with `Image`